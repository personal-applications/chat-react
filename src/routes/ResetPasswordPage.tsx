import useSWRMutation from "swr/mutation";
import ResetPasswordForm, {
  ResetPasswordFields,
} from "../forms/ResetPasswordForm";
import { postData } from "../services/fetcher";

function ForgotPasswordPage() {
  const { trigger } = useSWRMutation<
    { message: string },
    Error,
    "/api/auth/reset-password",
    ResetPasswordFields
  >("/api/auth/reset-password", postData);

  const onSubmit = async (data: ResetPasswordFields) => {
    const { message } = await trigger(data);

    return { message };
  };

  return (
    <>
      <div className="h-screen w-screen bg-neutral pt-40">
        <div className="w-[600px] mx-auto">
          <h1 className="text-3xl font-semibold text-center mb-5">ChatHub</h1>
          <h1 className="text-bold text-center text-xl"> Reset password </h1>
          <ResetPasswordForm onSubmit={onSubmit} />
        </div>
      </div>
    </>
  );
}

export default ForgotPasswordPage;
