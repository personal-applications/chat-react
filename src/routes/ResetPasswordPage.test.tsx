import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import useSWRMutation from "swr/mutation";
import { Mock } from "vitest";
import ResetPasswordPage from "./ResetPasswordPage";

vi.mock("swr/mutation", () => ({
  default: vi.fn().mockReturnValue({}),
}));

test("should render the reset password page", () => {
  const tree = render(
    <BrowserRouter>
      <ResetPasswordPage />
    </BrowserRouter>,
  );

  expect(tree).toMatchSnapshot();
});

test("should trigger the reset password mutation on form submission", async () => {
  const triggerMock = vi.fn().mockResolvedValue({ message: "Success" });
  (useSWRMutation as Mock).mockReturnValue({ trigger: triggerMock });

  render(
    <MemoryRouter initialEntries={["/reset-password?token=token"]}>
      <ResetPasswordPage />
    </MemoryRouter>,
  );

  const passwordInput = screen.getByPlaceholderText("New Password");
  const confirmPasswordInput = screen.getByPlaceholderText("Confirm Password");
  const submitButton = screen.getByRole("button", { name: "Reset" });

  await userEvent.type(passwordInput, "Password123*");
  await userEvent.type(confirmPasswordInput, "Password123*");
  await userEvent.click(submitButton);

  await waitFor(() => {
    expect(triggerMock).toHaveBeenCalledTimes(1);
    expect(triggerMock).toHaveBeenCalledWith({
      newPassword: "Password123*",
      confirmPassword: "Password123*",
      token: "token",
    });
    expect(screen.getByText("Success")).toBeInTheDocument();
  });
});
