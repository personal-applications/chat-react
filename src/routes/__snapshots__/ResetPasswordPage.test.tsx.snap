// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the reset password page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-screen w-screen bg-neutral pt-40"
      >
        <div
          class="w-[600px] mx-auto"
        >
          <h1
            class="text-3xl font-semibold text-center mb-5"
          >
            ChatHub
          </h1>
          <h1
            class="text-bold text-center text-xl"
          >
             Reset password 
          </h1>
          <form
            class="mt-6 flex flex-col gap-y-6 rounded-md bg-base-100 p-8"
            novalidate=""
          >
            <div
              class="flex w-full flex-col"
            >
              <label
                class="input input-bordered flex items-center gap-2"
              >
                <input
                  class="grow"
                  name="newPassword"
                  placeholder="New password"
                  type="password"
                />
                <button
                  type="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M396 512a112 112 0 10224 0 112 112 0 10-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                    />
                  </svg>
                </button>
              </label>
            </div>
            <div
              class="flex w-full flex-col"
            >
              <label
                class="input input-bordered flex items-center gap-2"
              >
                <input
                  class="grow"
                  name="confirmPassword"
                  placeholder="Confirm Password"
                  type="password"
                />
                <button
                  type="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M396 512a112 112 0 10224 0 112 112 0 10-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                    />
                  </svg>
                </button>
              </label>
            </div>
            <a
              class="link link-primary link-hover"
              href="/forgot-password"
            >
              Forgot password?
            </a>
            <button
              class="btn btn-primary"
              type="submit"
            >
              Reset
            </button>
          </form>
          <div
            class="flex flex-col gap-y-2 mt-3"
          >
            <span
              class="text-center"
            >
              Remember It ?
               
              <a
                class="link link-primary link-hover"
                href="/login"
              >
                Login
              </a>
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-screen w-screen bg-neutral pt-40"
    >
      <div
        class="w-[600px] mx-auto"
      >
        <h1
          class="text-3xl font-semibold text-center mb-5"
        >
          ChatHub
        </h1>
        <h1
          class="text-bold text-center text-xl"
        >
           Reset password 
        </h1>
        <form
          class="mt-6 flex flex-col gap-y-6 rounded-md bg-base-100 p-8"
          novalidate=""
        >
          <div
            class="flex w-full flex-col"
          >
            <label
              class="input input-bordered flex items-center gap-2"
            >
              <input
                class="grow"
                name="newPassword"
                placeholder="New password"
                type="password"
              />
              <button
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 1024 1024"
                  width="1em"
                >
                  <path
                    d="M396 512a112 112 0 10224 0 112 112 0 10-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                  />
                </svg>
              </button>
            </label>
          </div>
          <div
            class="flex w-full flex-col"
          >
            <label
              class="input input-bordered flex items-center gap-2"
            >
              <input
                class="grow"
                name="confirmPassword"
                placeholder="Confirm Password"
                type="password"
              />
              <button
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 1024 1024"
                  width="1em"
                >
                  <path
                    d="M396 512a112 112 0 10224 0 112 112 0 10-224 0zm546.2-25.8C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM508 688c-97.2 0-176-78.8-176-176s78.8-176 176-176 176 78.8 176 176-78.8 176-176 176z"
                  />
                </svg>
              </button>
            </label>
          </div>
          <a
            class="link link-primary link-hover"
            href="/forgot-password"
          >
            Forgot password?
          </a>
          <button
            class="btn btn-primary"
            type="submit"
          >
            Reset
          </button>
        </form>
        <div
          class="flex flex-col gap-y-2 mt-3"
        >
          <span
            class="text-center"
          >
            Remember It ?
             
            <a
              class="link link-primary link-hover"
              href="/login"
            >
              Login
            </a>
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
